### DISCUSSION TITLE
How can I access uResolution in fragmentSource?

### DISCUSSION THREAD

**Discussion by SSShooter** [fabricjs/fabric.js#9773] [Q&A] [answered]
I want to use a fragment shader which use `u_resolution`.

**Comment by asturur** [comment, marked as answer]
where does u_resolution comes from?

We have uStepW and uStepH
`
  uniform float uStepW;
  uniform float uStepH;
`
That tells you ( if i  recall correctly ) how much percentage of the image each pixel is

**Reply by SSShooter** [reply]
Great! that's what I need! Thank you so much😆
